app/connector/support/maven-plugin/src/main/java/io/syndesis/connector/support/maven/GenerateConnectorInspectionsMojo.java

```diff
@@ -102,6 +117,78 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
     }
 
+    private void initJsonSchema() throws MojoExecutionException {
+        if (jsonSchema != null) {
+            return; // already initialised
+        }
+
+        InputStream schemaStream = null;
+        try {
+            schemaStream = getClass().getResourceAsStream("/" + CONNECTOR_SCHEMA_FILE);
+            if (schemaStream == null) {
+                throw new IOException("Schema Initialisation Error: file not available");
+            }
+
+            LoadingConfiguration loadingConfiguration = LoadingConfiguration.newBuilder()
+                .dereferencing(Dereferencing.INLINE).freeze();
+            JsonSchemaFactory factory = JsonSchemaFactory.newBuilder()
+                .setLoadingConfiguration(loadingConfiguration).freeze();
+
+            JsonNode schemaObject = Json.reader().readTree(schemaStream);
+            jsonSchema = factory.getJsonSchema(schemaObject);
+        } catch (IOException | ProcessingException ex) {
+            throw new MojoExecutionException("Schema Initialisation Error", ex);
+        } finally {
+            if (schemaStream != null) {
+                try {
+                    schemaStream.close();
+                } catch (IOException e) {
+                    getLog().error(e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Validate the given file against the connection schema
+     *
+     * @param jsonStream {@link InputStream} of the file to validate.
+     *                                          Method will close the stream after use.
+     * @return the {@link JsonNode} of the file content or throw exception if an error
+     * @throws MojoExecutionException
+     */
+    public JsonNode validateWithSchema(String fileName, InputStream jsonStream) throws MojoExecutionException {
+        initJsonSchema();
+
+        try {
```

Comment:
If `validateWithSchema` receives a `File` instead of `InputStream` we can do try-with-resources here.

Reply:
See above comment about code validations post-compile.

Reply:
Let me checkout the branch and have a look, we use try-with-resources all over the place, should work...

Reply:
Yeah I see the same thing. The root cause seems to be https://github.com/spotbugs/spotbugs/issues/756, using Java 8 doesn't help either. Let's merge this and keep an eye on that issue and revisit when/if fixed...